compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -481,6 +481,25 @@ TR::Instruction *fixedSeqMemAccess(TR::CodeGenerator *cg, TR::Node *node, intptr
          nibbles[idx] = cursor = generateMemSrc1Instruction(cg, opCode, node, memRef, srcOrTrg, cursor);
       }
 
+   // When using tempReg ensure no register spills occur in the middle of the fixed sequence
+   if (tempReg)
+      {
+      TR::RegisterDependencyConditions *dep;
+      if (srcOrTrg == tempReg || srcOrTrg == baseReg)
+         {
+         dep = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 2, cg->trMemory());
+         }
+      else
+         {
+         dep = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 3, cg->trMemory());
+         dep->addPostCondition(srcOrTrg, TR::RealRegister::NoReg);
+         }
```

Comment:
you can use length-3 dependency-condition regardless whether srcOrTrg needs explicitly to be added into. code that way looks simpler and shorter, otherwise LGTM.

Reply:
Agree, fixed in f14cd51